kedro/io/data_catalog.py

```diff
@@ -288,6 +288,12 @@ class to be loaded is specified with the key ``type`` and their
         user_default = {}
 
         for ds_name, ds_config in catalog.items():
+            if not isinstance(ds_config, dict):
```

Comment:
In https://github.com/kedro-org/kedro/issues/3555 there was a bit of discussion on what to do with values that are dict type, but not meant as dataset. Is it worth updating the message for that case as well to mention interpolation?

Reply:
Yep, I've added a hint to the error that comes from `AbstractDataset.from_config()` as well!